src/styles/main.scss

```diff
@@ -0,0 +1,60 @@
+html {
+  margin: 0;
+  width: 100%;
+  height: 100%;
+  display: flex;
+  justify-content: center;
+  align-items: center;
+  font-family: Arial, sans-serif;
+}
+
+.calendar {
+  display: flex;
+  box-sizing: border-box;
+  flex-flow: row wrap;
+  width: $calendar-width;
+
+  &__day {
+    width: $day-width;
+    height: $day-height;
+    margin: calc(#{$day-gap} / 2);
+    box-sizing: border-box;
+    background-color: $day-bg-color;
+    border: 1px solid $border-color;
+    transition: 0.5s;
+    position: relative;
+
+    &:hover {
+      background-color: $day-hover-bg-color;
+      transform: translateY(-20px);
+      cursor: pointer;
+    }
+
+    @for $day from 1 through $days-count {
+      &:nth-child(#{$day})::before {
+        content: '#{$day}';
+        font-size: $day-font-size;
+        position: absolute;
+        left: 50%;
+        top: 50%;
+        transform: translate(-50%, -50%);
+      }
+    }
+  }
+
+  @for $i from 28 through $days-count {
+    &--month-length-#{$i} {
+      .calendar__day:nth-child(n + #{$i + 1}) {
+        display: none;
+      }
+    }
+  }
```

Comment:
The use of the `@for` directive here is correct, but ensure that `$days-count` is defined and holds the correct value for the number of days in the month. This loop generates classes for each possible month length from 28 to `$days-count` days.